src/meta-types/object.ts

```diff
@@ -152,13 +172,17 @@ export type ResolveObject<
   Deserialized<META_OBJECT>,
   DeepMergeUnsafe<
     If<
-      IsObjectOpen<META_OBJECT>,
+      ObjectCloseOnResolve<META_OBJECT>,
+      {},
       If<
-        IsObjectEmpty<META_OBJECT>,
-        { [KEY: string]: Resolve<ObjectOpenProps<META_OBJECT>, OPTIONS> },
-        { [KEY: string]: Resolve<Any, OPTIONS> }
-      >,
-      {}
+        IsObjectOpen<META_OBJECT>,
+        If<
```

Comment:
I handled `closeOnResolve` and `openProps` independently until type resolution.

If viable, another approach could have consisted of setting `openProps` to `Never` when `closeOnResolve` is true.